src/plat/qemu-arm-virt/config.cmake

```diff
@@ -12,18 +12,18 @@ declare_platform(qemu-arm-virt KernelPlatformQEMUArmVirt PLAT_QEMU_ARM_VIRT Kern
 set(qemu_user_top 0xa0000000)
 
 macro(setup_qemu_armv7)
-    if("${KernelSel4Arch}" STREQUAL aarch32)
-        declare_seL4_arch(aarch32)
-    elseif("${KernelSel4Arch}" STREQUAL arm_hyp)
-        declare_seL4_arch(arm_hyp)
-    else()
-        fallback_declare_seL4_arch_default(aarch32)
-    endif()
-    if(KernelSel4ArchArmHyp)
-        set(qemu_user_top 0xe0000000)
-    endif()
+    cmake_parse_arguments(ARMV7_OPTIONS "ve" "" "")
     set(QEMU_ARCH "arm")
     set(KernelArchArmV7a ON)
+    if(ARMV7_OPTIONS_ve)
+        declare_seL4_arch(aarch32 arm_hyp)
+        set(KernelArchArmV7ve ON)
```

Comment:
I don't see any users of either `KernelArchArmV7ve` or `ARCH_ARM_V7VE`, so it could be removed altogether if we want.

Reply:
It seems we could clean up some things here, but that should be a separate PR then.

Reply:
Agreed.

Reply:
It turns out that `ARCH_ARM_V7VE` seem unused in the C code, but `KernelArchArmV7ve` is used implicitly when we invoke the compiler with `-march=...` then. We might be able to clean up things a bit given that ARMv7 is superseded by now and things will never change beyond what we currently do. But we can't remove this completely.